vyper/compiler/output.py

```diff
@@ -132,7 +134,19 @@ def _compress_source_map(code, pos_map, jump_map, source_id):
     return compressed_map
 
 
+# EIP-170. Ref: https://eips.ethereum.org/EIPS/eip-170
+EIP_CONTRACT_SIZE_LIMIT: int = 2 ** 14 + 2 ** 13
+
+
 def build_bytecode_output(compiler_data: CompilerData) -> str:
+    compiled_bytecode_length = len(compiler_data.bytecode)
+    if compiled_bytecode_length > EIP_CONTRACT_SIZE_LIMIT:
+        warnings.warn(
+            f"Length of compiled bytecode is bigger than Ethereum contract size limit: "
```

Comment:
A specific reference to EIP170 would be good here, so users can more easily search out information about the limit.